src/ImageStack.cpp

```diff
@@ -50,75 +51,104 @@ int ImageStack::addImage(Image && i) {
 }
 
 
-void ImageStack::calculateSaturationLevel(const RawParameters & params, bool useCustomWl) {
-    // Calculate max value of brightest image and assume it is saturated
-    Image& brightest = images.front();
+void ImageStack::calculateSaturationLevel(const RawParameters & params, bool use_custom_white_level) {
+
+    /*
+     * This method will approximate a saturation threshold for the brightest
+     * image only. It is sufficient because the calculated threshold will
+     * be translated to the rest of the images in the stack in a relative
+     * manner thanks to each image's response function being pre-calculated.
+     *
+     * The saturation threshold has a major role in deciding on which pixels to
+     * use from which image. The better this value is calculated the better the
+     * results.
+     *
+     * Methodology Overwiew:
+     *  1. Generate the 4 histograms (green, blue, green, red) from the
+     *     brightest image
+     *  2. Determine the brightest value per histogram but ignore the outliers
+     *     using the threshold `occurance_threshold` that will discard bright
+     *     pixels that don't have a significant frequency (occurance)
+     *  3. Use vorious attempts to enhance the `saturation_threshold`
+     */
 
-    std::vector<std::vector<size_t>> histograms(4, std::vector<size_t>(brightest.getMax() + 1));
+    Image& brightest_image = images.front();
+
+    std::vector<std::vector<size_t>> histograms(4, std::vector<size_t>(brightest_image.getMax() + 1));
```

Comment:
I have no idea, I just read a bit online to find out the diff. `std::array` will be stored on the stack. wouldn't this be a problem since 4 histograms are gonna be relatively Lange? 🤔

Reply:
The histogram data will still be on the heap, as it's still in those four `std::vector<>`s which in turn will be stored in the `std::array<>`. In this case it's more a matter of fixed vs. variable size.

HTH,
Flössie

Reply:
Ok so something like this ?
``` c++
#include <array>
//...
std::array<std::vector<size_t>, 4> histograms;
std::fill(histograms.begin(), histograms.end(), std::vector<size_t>(brightest_image.getMax() + 1));
```

Reply:
Exactly. There is even a [`std::array<>::fill()`](https://en.cppreference.com/w/cpp/container/array/fill), I wasn't aware of. This will make it shorter.